hummingbot/market/bitfinex/bitfinex_market.pyx

```diff
@@ -0,0 +1,1161 @@
+                                                      decimal_price,
+                                                      order_id))
+        except asyncio.CancelledError:
+            raise
+        except Exception:
+            self.c_stop_tracking_order(order_id)
+            order_type_str = "MARKET" if order_type == OrderType.MARKET else "LIMIT"
+            self.logger().network(
+                f"Error submitting buy {order_type_str} order to Bitfinex for "
+                f"{decimal_amount} {trading_pair} {price}.",
+                exc_info=True,
+                app_warning_msg="Failed to submit buy order to Bitfinex. "
+                                "Check API key and network connection."
+            )
+            self.c_trigger_event(self.MARKET_ORDER_FAILURE_EVENT_TAG,
+                                 MarketOrderFailureEvent(self._current_timestamp,
+                                                         order_id, order_type))
+
+    cdef c_stop_tracking_order(self, str order_id):
+        """
+        Delete an order from self._in_flight_orders mapping
+        """
+        if order_id in self._in_flight_orders:
+            del self._in_flight_orders[order_id]
+
+    # sell
+    cdef str c_sell(self,
+                    str trading_pair,
+                    object amount,
+                    object order_type=OrderType.MARKET,
+                    object price=s_decimal_0,
+                    dict kwargs={}):
+        """
+        *required
+        Synchronous wrapper that generates a client-side order ID and schedules the sell order.
+        """
+        cdef:
+            # TODO: в доках используют time.time() надо разобратсья почему тут умножение
+            int64_t tracking_nonce = <int64_t>(time.time() * 1e6)
+            str order_id = str(f"sell-{trading_pair}-{tracking_nonce}")
+        safe_ensure_future(self.execute_sell(order_id, trading_pair, amount, order_type, price))
+        return order_id
+
+    async def execute_sell(self,
+                           order_id: str,
+                           trading_pair: str,
+                           amount: Decimal,
+                           order_type: OrderType,
+                           price: Optional[Decimal] = s_decimal_0):
+        """
+        Function that takes strategy inputs, auto corrects itself with trading rule,
+        and submit an API request to place a sell order
+        """
+        cdef:
+            TradingRule trading_rule = self._trading_rules[trading_pair]
+
+        decimal_amount = self.quantize_order_amount(trading_pair, abs(amount))
+        decimal_price = self.quantize_order_price(trading_pair, price)
+        if decimal_amount < trading_rule.min_order_size:
+            raise ValueError(f"Sell order amount {decimal_amount} is lower than the minimum order size "
+                             f"{trading_rule.min_order_size}.")
+
+        try:
+            # sell - is negative amount
+            decimal_amount *= -1
```

Comment:
You might want to document this in the test case for `test_limit_sell` and `test_market_sell`